core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/http/AmazonAsyncHttpClient.java

```diff
@@ -169,6 +170,7 @@ public <OutputT> CompletableFuture<OutputT> execute(
                                 .then(ApplyUserAgentStage::new)
                                 .then(MergeCustomHeadersStage::new)
                                 .then(MergeCustomQueryParamsStage::new)
+                                .then(CompressRequestStage::new)
```

Comment:
What is the reason of adding it in 7th position and not in any other position ?

Reply:
its arbitrary, it should be able to go anywhere between `MakeRequestMutableStage` and `MakeRequestImmutableStage`